spikeinterface/core/old_api_utils.py

```diff
@@ -45,16 +45,171 @@ def get_channel_property(self, channel_id, property):
         v = values[ind[0]]
         return v
 
-
 def create_extractor_from_new_recording(new_recording):
     old_recording = NewToOldRecording(new_recording)
     return old_recording
 
+class NewToOldSorting:
+    """
+    This class mimic the old API of spikeextractors with:
+      * unique segment
+    """
+    extractor_name = 'NewToOldSorting'
+    is_writable = False
+
+    def __init__(self, sorting):
+        assert sorting.get_num_segments() == 1
+        self._sorting = sorting
+        self._sampling_frequency = sorting.get_sampling_frequency()
+        self.is_dumpable = False
+
+    def get_unit_ids(self):
+        """This function returns a list of ids (ints) for each unit in the sorsted result.
+
+        Returns
+        -------
+        unit_ids: array_like
+            A list of the unit ids in the sorted result (ints).
+        """
+        return self._sorting.get_unit_ids()   
```

Comment:
In the new API, unit ids can also be str (or any object), while in the old API they're expected to be int. So here we have to check: if they are int, propagate them, if they are not int then they should be reset to int

Reply:
@khl02007 I tried to fix the int issue. Let me know what you think!